ddtrace/tracer/textmap.go

```diff
@@ -917,3 +909,26 @@ func parseTracestate(ctx *spanContext, header string) error {
 	}
 	return nil
 }
+
+// extractTraceID128 extracts the trace id from v and populates the traceID
+// field, and the traceID128 field (if applicable) of the provided ctx,
+// returning an error if v is invalid.
+func extractTraceID128(ctx *spanContext, v string) error {
+	if len(v) > 32 {
+		v = v[len(v)-32:]
+	}
+	v = strings.TrimLeft(v, "0")
+	var err error
+	if len(v) <= 16 { // 64-bit trace id
+		ctx.traceID, err = strconv.ParseUint(v, 16, 64)
+	} else { // 128-bit trace id
+		id128 := v[:len(v)-16]
+		// pad ctx.traceID128 with zeroes to ensure length of 16
+		ctx.traceID128 = fmt.Sprintf("%016s", id128)
```

Comment:
I knew about this for integers, but didn't know you could pad strings like this! :100: